services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowPriorityRerouteSpec.groovy

```diff
@@ -0,0 +1,65 @@
+package org.openkilda.functionaltests.spec.flows
+
+
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.FlowPayload
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+
+import java.util.concurrent.TimeUnit
+
+class FlowPriorityRerouteSpec extends BaseSpecification {
+    def "System is able to reroute flow in the correct order based on the priority field"() {
+        given: "3 flows on the same path, with alt paths available"
+        def switches = topology.activeSwitches
+        List<List<PathNode>> allPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.unique { it.sort() }.find { Switch src, Switch dst ->
+            allPaths = database.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 2
+        }
+        List<FlowPayload> flows = []
+
+        def newPriority = 100
+
+        3.times {
+            def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+            flow.maximumBandwidth = 10000
+            newPriority += 100
```

Comment:
I would expect this to be incremented _after_ L33, so that you have 100,200,300 priorities, and not 200, 300, 400.

Reply:
Done